moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -366,6 +366,12 @@ void ServoCalcs::calculateSingleIteration()
   // After we publish, status, reset it back to no warnings
   status_ = StatusCode::NO_WARNING;
 
+  // original_joint_state_ contains state x(t - 1)
+  // internal_joint_state_ will be updated with the state x(t + dt) in this iteration.
+  // last_joint_state_ will preserve the state x(t - dt) for this iteration to be used in central difference.
+  // original_joint_state_ will get updated with current state x(t) in updateJoints()
```

Comment:
I appreciate the explanation here!

I think you should do the caching at the end of each iteration, though. That seems to flow better. Can you move this code to L513 instead?

```
  // Cache the state for a finite difference calculation in the next iteration
  // last_joint_state_ = original_joint_state_;
```